auto_submit/test/service/config_test.dart

```diff
@@ -0,0 +1,51 @@
+// Copyright 2020 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:typed_data';
+
+import 'package:appengine/appengine.dart';
+import 'package:auto_submit/service/config.dart';
+import 'package:auto_submit/service/secrets.dart';
+import 'package:neat_cache/cache_provider.dart';
+import 'package:neat_cache/neat_cache.dart';
+import 'package:test/test.dart';
+
+/// Number of entries allowed in [Cache].
+const int kCacheSize = 1024;
+void main() {
+  group('Config', () {
+    late CacheProvider cacheProvider;
+    late Config config;
+    const int kCacheSize = 1024;
+
+    setUp(() {
+      cacheProvider = Cache.inMemoryCacheProvider(kCacheSize);
+      config = Config(
+        cacheProvider: cacheProvider,
+        secretManager: CloudSecretManager(),
+      );
+    });
+
+    test('githubAppInstallationId ', () async {
+      await withAppEngineServices(() async {
```

Comment:
Avoid using `withAppEngineServices` in tests. Instead, we should inject the httpProvider into the config so it's testable

Reply:
Hi Casey, when I tried to delete withAppEngineServices and use httpProvider, I got an error: Bad state: Not running inside a service scope zone.
I think this is because it failed to look up to get authClientService. Do you have any suggestions on this error?

Reply:
The test is incorrectly using `CloudSecretManager`. This is for prod workflows only. You'll want `LocalSecretManager()` for tests

Reply:
Oh I see, gotcha, thanks Casey!